integrations/pr_publish.py

```
@@ -0,0 +1,757 @@
+    # The char class permits ``.`` (repo names may contain dots), so also reject a ``.``/``..``
+    # *segment* explicitly — otherwise "../x" or "o/.." would pass and traverse the request path.
+    if not _REPO_RE.match(repo) or any(part in (".", "..") for part in repo.split("/")):
+        raise ValueError("repo must be 'owner/name' with no path/query characters (got %r); "
+                         "set --repo or $GITHUB_REPOSITORY" % repo)
+    return repo
+
+
+def _resolve_sha(arg):
+    sha = (arg or "").strip()
+    if sha and not _SHA_RE.match(sha):
+        raise ValueError("--sha must be 7–64 hex characters (got %r); set --sha or $GITHUB_SHA" % sha)
```

Ruff (0.16.1)

[warning] String contains ambiguous `–` (EN DASH). Did you mean `-` (HYPHEN-MINUS)?

(RUF001)